### ISSUE TITLE
Shadow endpoint errors should allow me to specify the error message.

### ISSUE THREAD

**Issue by Karlinator** [sveltejs/kit#3715]
### Describe the problem

As far as I can tell, when returning a 400 (say) response from a shadow endpoint, there is no way to pass a custom error message to the error page.

I might want to do something like this:
```
if (!valid_combination) {
    return {
        status: 400,
        error: 'You have entered the wrong combination'
    }
}
```

But judging by https://github.com/sveltejs/kit/blob/89a97551f9e771e0e00064646a5196b943480ab1/packages/kit/src/runtime/server/page/load_node.js#L441 there is no way to pass an error message other than 'Failed to load data'.

### Describe the proposed solution

Basically,
```
 return {
    status: 400,
    error: 'Custom error message'
}
```

### Alternatives considered

I can return `200` with an error message in the body, which is unsemantic but works (and bypasses the error page completely).

I can use a redirect to an error page (that isn't an 'error page').

I can go back to using a `load` function, from where I can do precisely this.

### Importance

would make my life easier

**Comment by florianlouvet** [comment]
This should work already. I've been testing this yesterday with no problem displaying back the error message I wrote in the endpoint.

```javascript
export async function post({ request }) {
	const data = await request.formData();
	if (!data.get("type")) return { status: 400, body: { error: "missing type of request" } };
	return { status: 400, body: { error: "invalid type of request" } };
};
```

With a sample Svelte route:
```svelte
<script>
    export let error = null;
</script>
{#if error}
    <p>{error}</p>
{/if}
<form method="post">
    <button type="submit">Submit</button>
</form>
```

**Comment by Karlinator** [comment]
I mean for e.g GET requests, where you are sent to an `__error.svelte` page. POST requests specifically send you back to the same page you started on (and, in fact, seems to ignore the status code but that's another matter) so you can display form validation errors, but if a shadow GET handler returns a 400 you end up on an `__error.svelte` page with no way to send it custom messages.

**Comment by georgecrawford** [comment]
Yep, I 100% agree that the error message is very important data to retain and display in some situations. Indeed, in local dev, I'd like access to the full Error include stacktrace. You can see that with a [tiny change](https://github.com/sveltejs/kit/pull/4032/files), we can expose this message for both client-side and server-side rendering. Sending a stacktrace would require serializing the error object (see PR comments), and should probably be avoided in non-development environments, but I'd love that too!

**Comment by Rich-Harris** [comment]
As of https://github.com/sveltejs/kit/pull/5314 you can pass an `Error` instance as the body, and it will be used to render the `__error.svelte` page, so I'll close this
